newm-server/src/main/kotlin/io/newm/server/auth/jwt/JwtConfig.kt

```diff
@@ -7,37 +7,60 @@ import io.ktor.server.auth.AuthenticationConfig
 import io.ktor.server.auth.jwt.JWTPrincipal
 import io.ktor.server.auth.jwt.jwt
 import io.newm.server.auth.jwt.repo.JwtRepository
+import io.newm.server.ktx.getSecureConfigString
 import io.newm.shared.koin.inject
 import io.newm.shared.ktx.getConfigString
 import io.newm.shared.ktx.toUUID
+import io.newm.shared.ktx.warn
+import kotlinx.coroutines.runBlocking
+import org.koin.core.parameter.parametersOf
+import org.slf4j.Logger
 
 const val AUTH_JWT = "auth-jwt"
 const val AUTH_JWT_REFRESH = "auth-jwt-refresh"
+const val AUTH_JWT_ADMIN = "auth-jwt-admin"
 
 fun AuthenticationConfig.configureJwt() {
     configureJwt(AUTH_JWT, JwtType.Access)
     configureJwt(AUTH_JWT_REFRESH, JwtType.Refresh)
+    configureJwt(AUTH_JWT_ADMIN, JwtType.Access)
 }
 
 private fun AuthenticationConfig.configureJwt(name: String, type: JwtType) {
     val environment: ApplicationEnvironment by inject()
     val repository: JwtRepository by inject()
+    val logger: Logger by inject { parametersOf(javaClass.simpleName) }
 
     jwt(name) {
         this.realm = environment.getConfigString("jwt.realm")
         verifier(
-            JWT.require(Algorithm.HMAC256(environment.getConfigString("jwt.secret")))
+            JWT.require(Algorithm.HMAC256(runBlocking { environment.getSecureConfigString("jwt.secret") }))
                 .withAudience(environment.getConfigString("jwt.audience"))
                 .withIssuer(environment.getConfigString("jwt.issuer"))
                 .withClaim("type", type.name)
+                .apply {
+                    if (name == AUTH_JWT_ADMIN) {
+                        withClaim("admin", true)
+                    }
+                }
                 .build()
         )
         validate { credential ->
             with(credential) {
-                payload.id?.takeIf { jwkId ->
-                    repository.exists(jwkId.toUUID())
-                }?.let {
+                val admin = payload.getClaim("admin")?.asBoolean() == true
+                val exists = payload.id?.let { jwkId -> repository.exists(jwkId.toUUID()) } == true
+
+                val isValid = if (name == AUTH_JWT_ADMIN) {
+                    admin && exists
+                } else {
+                    exists
+                }
```

Comment:
I think that we don't need to check the admin claim here since we are already enforcing it with the Verifier in line 43. It's not different from the way that we enforce the other claims like audience (line 38) and issuer (line 39) and type line 40).

Reply:
While that's true, I'd feel more safe having both checks in the code. It forces an attacker to get past two locked doors instead of 1.